src/library/automatic_oa_damper_controls.py

```diff
@@ -1,12 +1,24 @@
 from checklib import RuleCheckBase
 
+import numpy as np
+
 
 class AutomaticOADamperControl(RuleCheckBase):
-    points = ["o", "m_oa", "eco_onoff", "tol"]
+    points = ["o", "eco_onoff", "m_oa", "m_ea", "tol_o", "tol_m_oa", "tol_m_ea"]
+
+    def automatic_oa_damper_check(self, data):
+        if data["o"] < data["tol_o"]:
+            if data["eco_onoff"] == 0 and (
+                data["m_oa"] >= data["tol_m_oa"] or data["m_ea"] >= data["tol_m_ea"]
+            ):
+                return False
+            else:
+                return True
+        else:
+            return np.nan
 
     def verify(self):
-        self.result = ~(
-            (self.df["o"] < self.df["tol"])
-            & (self.df["m_oa"] > 0)
-            & (self.df["eco_onoff"] == 0)
+        self.df["result"] = self.df.apply(
+            lambda d: self.automatic_oa_damper_check(d), axis=1
         )
+        self.result = self.df["result"]
```

Comment:
maybe directly assign to self.result without adding a column to df so there is no additional point in the plots later